### PR TITLE
Feat/handle emojis GitHub

### LINKED ISSUE #36: Emoji in header breaks generated link
Continued from https://github.com/thlorenz/doctoc/issues/123

In short: **If you place an emoji into a header, the generated anchor tag does not work.**

----

The actual link generated by GitHub just leaves out the emoji, but has a dash in there to show the space.

So I actually tried it out with a heading like this:

```
# Modules 📦
```

And the Markdown required is:

```
- [Modules](#modules-)
```

You can see it here: https://github.com/adrianmcli/next-boilerplate/blob/master/README.md

### PR DESCRIPTION
**Pull request by adrianmcli**
Fixes #36 

I added 3 tests and a dependency on [emoji-regex](https://github.com/mathiasbynens/emoji-regex).

### CODE REVIEW

**Comment by thlorenz** [comment]
Please remove the yarn stuff, that has nothing to do with the fix you're trying to do.
Also that seemed to have broken the build.

If you try to add yarn please file a PR (even though I'll refuse it). In general please keep PRs focused on one thing.

Thanks

**Comment by thlorenz** [comment]
After you fixed the yarn thing please squash all commits and push.

File: package.json [hunk 1]
```diff
@@ -24,5 +24,8 @@
   "gitHead": "dcde154040d17d321664a7efc38e8531330335f9",
   "devDependencies": {
     "tap": "~0.4.8"
+  },
+  "dependencies": {
+    "emoji-regex": "^6.1.0"
```
Comment (thlorenz) [inline package.json]: Please use `~` as range specifier here.
I'm conservative with deps and also this avoids breaking older npms which might be the reason the build is breaking on travis right now.

File: test/anchor-markdown-header.js [hunk 2]
```diff
@@ -43,6 +43,9 @@ test('\ngenerating anchor in github mode', function (t) {
   , [ 'module-specific-variables-using-jsdoc-@module', null, '#module-specific-variables-using-jsdoc-module']
   , [ 'Jack & Jill', null, '#jack--jill']
   , [ 'replace – or —', null, '#replace--or-']
+  , [ 'Modules 📦', null, '#modules-']
```
Comment (thlorenz) [inline test/anchor-markdown-header.js]: Is this correct to have a trailing `-` here?
Have you tried if github does it that way?

Comment (adrianmcli) [inline test/anchor-markdown-header.js]: Yes it is.

File: package.json (hunk 1 above)
Comment (adrianmcli) [inline package.json]: sure

**Comment by adrianmcli** [comment]
@thlorenz I'm having trouble squashing the commits since this branch has already been pushed to the PR. You should be able to squash it before you merge.

squashing

**Comment by adrianmcli** [comment]
Here's a better example picture:

squash and merge

**Comment by thlorenz** [comment]
Sorry please squash at the command line via an interactive rebase.
Don't use the horrible UI.

**Comment by adrianmcli** [comment]
There, done.

**Comment by thlorenz** [comment]
Awesome, thanks now I made you collaborator.

Please merge to master following [this guide](https://gist.github.com/thlorenz/f7df0daf13c6f6c1fa84).
I will then version and publish to npm.

Thanks a lot for your help.

I'll version as a patch so doctoc will automatically pick up the fix when you install it fresh.

**Comment by thlorenz** [comment]
@adrianmcli when I meant I don't want the history, I meant the commit message as well, i.e:

```
add yarn lock file

remove yarn file
```

is completely nonsensical.  To me it looks like you just squashed without editing the commit message at all.
I'll fix that and force push to master, but please submit meaningful and to the point commit messages in the future.

I'll also publish as a patch to npm.

**Comment by thlorenz** [comment]
[merged](https://github.com/thlorenz/anchor-markdown-header/commit/3ab80f2a697db5beb5225a4612538849352ac028) and ublished as 0.5.7.

**Comment by adrianmcli** [comment]
Sorry about that, I thought the whole point of the squashing was to document the history of what happened without multiple commits. I'll keep this in mind in the future.

**Comment by thlorenz** [comment]
The whole point of squashing is to erase the history we don't need to know, i.e. that you added a yarn file and then removed it.

A commit message should state what the commit does, not how you got there, but no worries, just wanted to let you know, so you can do it better next time :)

**Pull request closed without merging**